### ISSUE TITLE
Local Adjustments: Horizontal field size too large

### ISSUE THREAD

**Issue by rebio** [Beep6581/RawTherapee#6153]
**Short description**
The horizontal size of some elements of the local adjustments is too large and requires horizontal scrolling even there is no need for it (unless sliders have a fixed size)

**Steps to reproduce**
1. Open any image
2. Activate Local Adjustments 
3. Add a new area of effect
4. Add a "Shadows/Highlights & Tone equalizer" section (others like "Common Color Mask" will also work; some like "Color & Light" or "Sharpening" will not cause this)

Side Notes:
1. When you start RT and you do not select an image and go right to the LA-tab the size of the panel is also too large and needs to be scrolled
2. Is it maybe because of the name of the single setting? Maybe the width of the panel is set by the width of the text of the single setting. I'd rather have a small box where I do not have to scroll and maybe have to hover over the text to see the full name.

**Expected behavior**
I would expect the panel to keep the original size. The problem is not only that you have to scroll but also that it's much harder to set the values using the slider because at some point the slider will _jump_. Therefore fine tuning by using the sliders is pretty much impossible.

**Additional information**
RT:
> Version: 5.8-2802-g84751ad26 (2021-03-01 - win64)
> Branch: dev
> Commit: 84751ad26
> Commit date: 2021-03-01
> Compiler: x86_64-w64-mingw32-gcc\ 9.3.0
> Processor: generic x86
> System: Windows
> Bit depth: 64 bits
> Gtkmm: V3.24.3
> Lensfun: V0.3.2.0
> Build type: Release
> Build flags: -mwin32\ -m64\ -mthreads\ -msse2\ -std=c++11\ -mtune=generic\ -Werror=unused-label\ -Werror=delete-incomplete\ -fno-math-errno\ -Wno-attributes\ -Wall\ -Wuninitialized\ -Wcast-qual\ -Wno-deprecated-declarations\ -Wno-unused-result\ -Wunused-macros\ -fopenmp\ -Werror=unknown-pragmas\ -Wno-aggressive-loop-optimizations\ -DNDEBUG\ -O3\ -ftree-vectorize
> Link flags: -m64\ -mthreads\ -static-libgcc\ -mtune=generic\ -s\ -O3
> OpenMP support: ON
> MMAP support: ON
> Build OS: Linux 4.4.0-101-generic x86_64
> Build date: Mon, 01 Mar 2021 23:36:34 +0000 UTC
> Build epoch: 1614641794
> Build UUID: fe7b4455-0570-4fdf-95bd-0eb80bb80975

OS:
> CPU: AMD Ryzen 7 4800H with Radeon Graphics
> RAM: 32GB
> OS: Windows 10, 64-bit

Local Adjustments startup:
![Local Adjustments - clipped sliders startup](https://user-images.githubusercontent.com/10911084/110019215-7cbf2e80-7d28-11eb-88a1-4d5a98a03a13.png)

Local Adjustments with on image:
![Local Adjustments - clipped sliders](https://user-images.githubusercontent.com/10911084/110019245-88aaf080-7d28-11eb-9ceb-ac213851d9fa.png)
Note that when the "Shadows/Highlights & Tone equalizer" is deleted the panel size is restored to the size it was before.

**Comment by Desmis** [comment]
@rebio
With my configuration I have not this problem...
But, I am not at all a GUI specialist...And I don't know exactly what are all the parameters that interact on this phenomenon?
Of course the lenght ot the "label"...but may also be the size and type of the font

Still, "Shadows/Highlights & Tone  Equalizer" is the label that is the longest....I propose to suppress the "Tone", so the new label is  "Shadows/Highlights & Equalizer"

I will push a commit soon

Jacques

**Comment by rebio** [comment]
@Desmis / @Jacques

You could try to resize the side bar (make it smaller) and see if you can produce that issue. When I resize the side panel to fit the the longest text, I also have no problem.
However, I'm using the default size and I'm using RT on a 1080p monitor (which is probably one of the most used resolutions right now). But even on my 2560x1600 screen I get the same issue.

Even though I'm happy about every change to make this better, I don't think that renaming the panel is the way to go since there will be the same issue with other panels or languages that may require a longer name. Also using a short name like this would may make the option you want to use harder to find.

**Comment by Desmis** [comment]
Of course, if you reduce the size of the window the problem appears. It is a compromise between understanding and GUI.
Unless someone has another solution, I've changed the length of some labels...and I think that in this case, it's not a problem.

The problem is fairly simple to state, but difficult to solve...at least for me.

if someone has an idea on how to put labels (length, understanding...) I'm interested right away 

jacques

**Comment by rebio** [comment]
Yesterday I discovered that the expansion is not only based on single setting you can add, but also cause on some value you have to expand to change it.
Eg. Under "Mask and Merge" there is a slider called "Background color for luminance and color masks". When you unfold the "Mask and Merge" segment the side panel will enlarge. If you collapse it, the horizontal size returns to the default

(The slider above the red line)
![LA - sidebar setting](https://user-images.githubusercontent.com/10911084/110232690-7b0b8b80-7f1f-11eb-9336-a1ed3a47f7ee.png)

**Comment by Thanatomanic** [comment]
@rebio @Desmis Notwithstanding some long labels in Local adjustment, another solution would be to recognize that we simply need a larger default panel size. I think it is now only 300 px, which actually leaves some horizontal empty space when you use RT full-screen with a left sidebar and filmstrip, and when you have a typical 3:2 photo:
![image](https://user-images.githubusercontent.com/6567747/110234249-a2ffec80-7f29-11eb-9663-7c6cfdb04d42.png)

When you enlarge the right sidebar to reduce the empty space around the image, the texts in the sidebar all fit:
![image](https://user-images.githubusercontent.com/6567747/110234216-8368c400-7f29-11eb-98ae-a63ec59906ce.png)

**Comment by Desmis** [comment]
@Thanatomanic  @rebio 
The problem is not very simple, because how to make certain choices understandable. For example for « Shadows/highlight & Tone equalizer » , if, in order to get into the possible size we delete "Tone Equalizer" we risk to deprive the user of the best choice...

Of course I can reduce some labels without too much trouble, but this will have limits.

And for « Background color for luminance and color masks »,I can change with « Background color for luminance masks », because there is a tooltip.

There are probably several solutions :
* enlarge default panel size – I don't know where exactly default values are. I think it is toolPanelWidth = 400 in option.cc..but I am not sure..it can be increased to 450  @Thanatomanic , could you confirm.
* reduce the size of the Segoe font (default) to 8. This is the choice I made, independently of this problem, because I find it more aesthetic
jacques

**Comment by Desmis** [comment]
@Thanatomanic  @rebio 
I just create a Pull Request https://github.com/Beep6581/RawTherapee/pull/6160

I change the label "background color..."
I keep all last changes for labels (shadows/highlight-Equalizer)
I enlarge tool panel to 460 (from 400)...I just change to 465
I change the size of the box for "basic"..."advanced"

Jacques

**Comment by Desmis** [comment]
@Thanatomanic @rebio 
Others improvment to reduce left panel size:
* adapted box "complexity" to font size
*  set font size to 8 by default
* minimize (Gtk::PACK_SHRINK, k)  k minimum (for this part)
* change some labels in "mask and modifications" ..too long

But it needs that "options" will be "reset"....Otherwise ?

jacques

**Comment by Thanatomanic** [comment]
@Desmis I will comment more later this week, thanks for exploring the options 😃

**Comment by Desmis** [comment]
@rebio @Thanatomanic 
In last commit bd33cad79 I have made different arrangements to improve size right panel

* increase default size right panel to an "adapted" size depending on font size
* change when it is possible and with "sens" upper case to lower case
I think now the "compromise" is good...
My preference is with sizefont = 8

Jacques
